### ISSUE TITLE
Add tests for event page

### ISSUE THREAD

**Issue by michaelbukachi** [droidconKE/droidconKE2019App#56]
This is a sub issue for #53

**Comment by johnGachihi** [comment]
I can work on this

**Comment by wangerekaharun** [comment]
Feel free to work on it

**Comment by johnGachihi** [comment]
@wangerekaharun is it ok if I move all the Firebase Remote Config code from [EventsFragment](https://github.com/droidconKE/droidconKE2019App/blob/develop/app/src/main/java/com/android254/droidconke19/ui/events/EventFragment.kt) into a new repository. This will make it a lot easier to test the fragment

**Comment by wangerekaharun** [comment]
Yes its okay to do so. Do the same for all Remote Config code.. We were supposed to do so but you can take the task.

**Comment by johnGachihi** [comment]
Alright

**Comment by johnGachihi** [comment]
Also @wangerekaharun can I ask what the reason is for having the getter methods for the livedata in the viewmodels. Are they really necessary?

**Comment by wangerekaharun** [comment]
More of separation of concerns. Having one method doing the fetch and the other one doing the get response from the livedata. This also helps in orientation changes. Assuming you only have one method for fetching and updating the livedata, it will be called every time we have orientation change meaning livedata will be updated again. In the current approach, since the fetch methods are called when needed, the livedata will still have their values even on orientation change

**Comment by johnGachihi** [comment]
If instead of using the livedata getters, we accessed the livedata directly (like this: `myViewModel.myLiveData`) would it make a difference.

**Comment by wangerekaharun** [comment]
Why would you want it to be so?

**Comment by johnGachihi** [comment]
I feel like the getters are not necessary. The livedata properties could just be made public and accessed directly.

**Comment by wangerekaharun** [comment]
@michaelbukachi what's your take on this?

**Comment by michaelbukachi** [comment]
I usually make live data public. It's just easier since it reduces the amount of code written.

**Comment by wangerekaharun** [comment]
@johnGachihi you can go ahead and make the changes

**Comment by johnGachihi** [comment]
> @johnGachihi you can go ahead and make the changes

@wangerekaharun I've found that it's good practice to use `MutableLiveData` in viewmodels but expose the immutable `LiveData` to Activities and Fragments like you had done.
But I feel like [this example](https://github.com/android/plaid/blob/4523a37e76bb389ee78e5b3bc00124454527df70/dribbble/src/main/java/io/plaidapp/dribbble/ui/shot/ShotViewModel.kt#L45-L55) is more Kotlin-like than using getter methods

**Comment by wangerekaharun** [comment]
Yes we already had agreed on this as @michaelbukachi had pointed out
